src/helpers.ts

```diff
@@ -974,19 +1088,45 @@ export function updateProtocolPosition(protocol: LendingProtocol, market: Market
   market.save();
 }
 
-function computeProportionIdle(market: Market): BigInt {
-  const offset = pow10(market._indexesOffset);
-  if (market._idleSupply && market._idleSupply!.gt(BigInt.zero())) {
-    const totalP2PSupplied = market._p2pSupplyAmount.times(market._p2pSupplyIndex).div(offset);
-    const proportionIdle = market._idleSupply!.times(offset).div(totalP2PSupplied);
-    if (proportionIdle.gt(offset)) return offset;
-    return proportionIdle;
+export function updateP2PIndexesAndRates(
+  event: ethereum.Event,
+  market: Market,
+  __MATHS__: IMaths
+): void {
+  const rates: string[] | null = market.rates;
+  if (!rates) {
+    log.debug("Rates have not been initialized", []);
+    return;
+  }
+  if (rates.length < 4) {
+    log.debug(
+      "Rate error, tried to call updateP2PIndexesAndRates before initializing reserve rates",
+      []
+    );
+    return;
   }
-  return BigInt.zero();
-}
 
-export function updateP2PRates(market: Market, __MATHS__: IMaths): void {
-  const proportionIdle = computeProportionIdle(market);
+  // Will throw with Index out of Range if array is smaller than 4.
```

Comment:
```suggestion
```
Youre already checking this case before

Reply:
Yes, it's not a check, it's just a comment to remember it, because it is a TS syntax, but accessing an element outside of the array is throwing. I made the mistake multiple times, so I wrote it to not forget it.